### PR TITLE
Expose query state in ManagedQueryExecution

### PR DESCRIPTION
**Pull request by aalbu**

### CODE REVIEW

File: presto-main/src/main/java/io/prestosql/dispatcher/LocalDispatchQuery.java [hunk 1]
```diff
@@ -250,6 +251,14 @@ public QueryInfo getFullQueryInfo()
                 .orElseGet(() -> stateMachine.updateQueryInfo(Optional.empty()));
     }
 
+    @Override
+    public QueryState getState()
+    {
+        return tryGetQueryExecution()
+                .map(queryExecution -> queryExecution.getQueryInfo().getState())
```
Comment (sopel39) [inline presto-main/src/main/java/io/prestosql/dispatcher/LocalDispatchQuery.java:258]: obtaining query info is even more expensive than basic stats

Comment (aalbu) [reply presto-main/src/main/java/io/prestosql/dispatcher/LocalDispatchQuery.java:258]: Ugh.  `stateMachine.getState()`?

Comment (sopel39) [reply presto-main/src/main/java/io/prestosql/dispatcher/LocalDispatchQuery.java:258]: Yes. That one is cheap

Comment (aalbu) [reply presto-main/src/main/java/io/prestosql/dispatcher/LocalDispatchQuery.java:258]: Thanks for catching.

**Pull request merged**